src/pages/index.js

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import Link from 'next/link';
+import Router from 'next/router';
+import Head from '../components/head';
+import Search from '../components/search';
+import millify from '../lib/millify';
+import intcomma from '../lib/intcomma';
+import { getHome } from '../lib/api';
+
+class Index extends React.Component {
+  static async getInitialProps() {
+    return getHome({
+      fiscalYear: null
+    });
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = Object.assign({}, props);
+  }
+
+  handleChangeFiscalYear = async (value) => {
+    this.setState((state) => {
+      return {
+        fiscal_year: {
```

Comment:
magic state? initial state should have a rough or same structure as what the state updates pass to setState

Reply:
He's probably expecting that from the values returned on `getInitialProps`. If that's the case maybe `getInitialProps` should manage the response and return an explicit default with `{fiscal_year: {}}`?

Reply:
Also, camelCase... 😬